packages/catalog-client/src/CatalogClient.ts

```diff
@@ -24,24 +24,27 @@ import fetch from 'cross-fetch';
 import {
   AddLocationRequest,
   AddLocationResponse,
-  CatalogApi,
   CatalogEntitiesRequest,
   CatalogListResponse,
   DiscoveryApi,
 } from './types';
 
-export class CatalogClient implements CatalogApi {
+export class CatalogClient {
   private readonly discoveryApi: DiscoveryApi;
 
   constructor(options: { discoveryApi: DiscoveryApi }) {
     this.discoveryApi = options.discoveryApi;
   }
 
-  async getLocationById(id: String): Promise<Location | undefined> {
-    return await this.getOptional(`/locations/${id}`);
+  async getLocationById(
+    token: string | undefined,
```

Comment:
Let's add the token in a second optional options param instead, so that it's not a breaking change and easier to add more options in the future.

```ts
type RequestOptions = {
  token?: string
}

async getLocationById(id: string, options?: RequestOptions)
async getEntities(request?: CatalogEntitiesRequest, options?: RequestOptions)
// ... etc
```

@freben, +1?

Reply:
Sounds familiar to what we do in other places. The type would probably be named `CatalogRequestOptions` then.